### ISSUE TITLE
Investigate possible issues with /played and various time based commands

### ISSUE THREAD

**Issue by tehbeard** [tehbeard/BeardStat#58]
from dbo (lostwave):

Also, any way you can clean up the played command so it shows days, hours, minutes, seconds like it used to instead of just minutes. EDIT: Noticed that lastlogin, lastlogout and lastsession stats doesn't usually work.

Thanks.

**Comment by Lostwave** [comment]
I am using version 0.8.1 Snapshot.  B0002a2
Spogot 1.8

No idea why or how the last login etc isn't working.  There isn't really a pattern.

The numbers are just off.  Negative numbers sometimes.  I created a new world the other day, it says I last logged in:
1378719269	-> 2013-09-09 05:34:29
My main world last logged in (I logged in just now)
434965320      -> 1983-10-14 03:42:00

But for some users, its right on. Is there a newer build that is stable that I should try?

**Comment by tehbeard** [comment]
Try https://github.com/tehbeard/BeardStat/releases/tag/spigot-test-build build and see if the issue persists?
/played appears to be working for me, /laston doesn't seem to be returning good data, I'm looking into if it's pulling from spigot and that data is messed up.

**Comment by Lostwave** [comment]
I am using the spigot test build now.

Played works, but it shows seconds which isn't very readable.
Laston is also returning unixtime, not human readable time.  And the values are bad as you also noticed.

**Comment by tehbeard** [comment]
Is it showing no formatting whatsoever just a raw number?

If so This might be a case of the statistic type not being set.

**Comment by Lostwave** [comment]
>played lostwave
2015-02-18 12:41:21 [INFO] 4146599 total
2015-02-18 12:41:21 [INFO] __imported__ = 4034634
2015-02-18 12:41:21 [INFO] msk = 390
2015-02-18 12:41:21 [INFO] Survival = 2894
2015-02-18 12:41:21 [INFO] Calypso = 16501
2015-02-18 12:41:21 [INFO] Azeroth = 517
2015-02-18 12:41:21 [INFO] LostwaveMuseum = 1672
2015-02-18 12:41:21 [INFO] Factions = 4529
2015-02-18 12:41:21 [INFO] Westeros = 85391

>laston lostwave
2015-02-18 12:41:35 [INFO] lostwave
2015-02-18 12:41:35 [INFO] First on: 1323731289
2015-02-18 12:41:35 [INFO] Last on: 1392071083

**Comment by Lostwave** [comment]
In my Stats_statistic table:
StatisticID	Statistic	Name	Formatting
165		firstlogin	NULL	none
279		lastlogin	NULL	none
280		lastlogout	NULL	none
373		playedfor	NULL	none

**Comment by Lostwave** [comment]
Wow, I just fixed it.  Checked in your SQL scripts and found what those were supposed to be.  This has been annoying me for sooo long.  
It changed from:
>laston lostwave
2015-02-18 12:41:35 [INFO] lostwave
2015-02-18 12:41:35 [INFO] First on: 1323731289
2015-02-18 12:41:35 [INFO] Last on: 1392071083

To: 
>laston Lostwave
2015-02-18 13:10:36 [INFO] Lostwave
2015-02-18 13:10:36 [INFO] First on: Fri Jan 16 02:42:11 EST 1970
2015-02-18 13:10:36 [INFO] Last on: Fri Jan 16 21:41:11 EST 1970

Now it looks better.
BUT, the formatting doesn't work, based on the unixtime numbers above.
The First on should be 2011-12-12 18:08:09
last on should be: 2014-02-10 17:24:43

AND it rarely saves laston (lastlogin or lastlogout) times.  I logged on today, and there is no laston time for today.

value, statistic, worldname, from_unixtime(value)
1323731289, firstlogin, __global__, 2011-12-12 18:08:09
1307726154, firstlogin, __imported__, 2011-06-10 13:15:54
1814962516, lastlogin, Westeros, 2027-07-07 08:15:16
1492053020, lastlogin, Factions, 2017-04-12 23:10:20
1404332269, lastlogin, msk, 2014-07-02 16:17:49
1392071083, lastlogin, __global__, Global, 2014-02-10 17:24:43
1390603500, lastlogin, __imported__, 2014-01-24 17:45:00
120792257, lastlogin, Calypso, 1973-10-29 21:24:17
-1446576342, lastlogin, LostwaveMuseum,  NULL
-1541444280, lastlogin, Survival, NULL

**Comment by Lostwave** [comment]
I think in the Statistic Meta you need to multiply by 1000, your saved times don't include milliseconds, but using new date() expects them.  Thats why its 1970.

Not sure if that will help with the saving issue, but would definitely help with the format.

**Comment by tehbeard** [comment]
Good catch on the date, not sure what's caused the negative numbers though.

**Comment by Lostwave** [comment]
This appears fixed.  Its now logging logoffs, formatting is correct.
I will let you know in a few days if I see any issues.

Thanks!

**Comment by Lostwave** [comment]
Another minor request that goes with this.
Would you mind removing the Weeks section from the /played command.
100 Days 0 hours 32 mins 
is more useful than
14 wks 2 days 0 hours 32 mins
No biggie tho

**Comment by Lostwave** [comment]
Appears to be all fixed.  Thanks!
